### PR TITLE
Merge the agents in a single crate

### PR DESCRIPTION
**Pull request by TheElectronWill**

### CODE REVIEW

**Comment by AngeCyp** [comment]
Is it a ready for review PR or just a draft currently ?

**Comment by TheElectronWill** [comment]
ready :) otherwise I would have marked it as a Draft in GitHub.
But that can wait tomorrow ^^

File: alumet/src/agent.rs [hunk 1]
```diff
@@ -779,7 +779,7 @@ macro_rules! static_plugins {
     [] => {
         Vec::<$crate::plugin::PluginMetadata>::new()
     };
-    [$($x:path),*] => {
+    [$($x:path),* $(,)?] => {
```
Comment (AngeCyp) [inline alumet/src/agent.rs:782]: is it to allow trailing comma ?

File: alumet/src/agent.rs [hunk 2]
```diff
@@ -807,6 +807,18 @@ mod tests {
 
     use super::{AgentConfig, ENV_VAR};
 
+    #[test]
+    fn static_plugins_macro() {
```
Comment (AngeCyp) [inline alumet/src/agent.rs:811]: Maybe you could test using:
```rust
 let x = static_plugins![,];
```

or is it not allowed ?

File: app-agent/src/bin/relay_client.rs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+use clap::{Parser, Subcommand};
+
+type AppConfig = config::CommonArgs;
+
+fn main() {
+    let plugins = static_plugins![
+        plugin_rapl::RaplPlugin,
+        plugin_perf::PerfPlugin,
+        plugin_socket_control::SocketControlPlugin,
+        plugin_relay::client::RelayClientPlugin,
```
Comment (AngeCyp) [inline app-agent/src/bin/relay_client.rs:18]: Is it possible to make people using relay_client (or relay_server) unable to desactivate this plugin ?

File: app-agent/src/bin/relay_server.rs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+use app_agent::{
+    agent_util, init_logger,
+    options::{cli, config, AgentModifier},
+};
+use clap::{Parser, Subcommand};
+
+type AppConfig = config::CommonArgs;
+
+fn main() {
+    let plugins = static_plugins![plugin_relay::server::RelayServerPlugin, plugin_csv::CsvPlugin];
```
Comment (AngeCyp) [inline app-agent/src/bin/relay_server.rs:11]: lint is different from other files

File: alumet/src/agent.rs (hunk 1 above)
Comment (TheElectronWill) [reply alumet/src/agent.rs:782]: yep!

File: app-agent/src/bin/relay_client.rs (hunk 3 above)
Comment (TheElectronWill) [reply app-agent/src/bin/relay_client.rs:18]: you mean, via the config? why?

File: app-agent/src/bin/relay_server.rs (hunk 4 above)
Comment (TheElectronWill) [reply app-agent/src/bin/relay_server.rs:11]: the formatting? It follows the rules of `rustfmt`, just like the other files. The line is short enough here.

File: alumet/src/agent.rs (hunk 2 above)
Comment (TheElectronWill) [reply alumet/src/agent.rs:811]: good idea

File: app-agent/src/bin/relay_client.rs (hunk 3 above)
Comment (AngeCyp) [reply app-agent/src/bin/relay_client.rs:18]: To prevent user from desactivating the plugin by error

**Review by titouanj — APPROVED** [review]

**Pull request merged**